hydroengine_service/dgds_functions.py

```diff
@@ -161,23 +159,106 @@ def visualize_gebco(source, band):
+                image = resample_landmask(image)
+            else:
+                image = image.float().resample('bicubic')
+            images.append(image)
+        elif type == 'ImageCollection':
+            image_collection = ee.ImageCollection(source)
+            image_collection = image_collection.map(resample_landmask)
+            image_collections.append(image_collection)
+
+    dems = ee.ImageCollection(images)
+    for collection in image_collections:
+        dems = dems.merge(collection)
+
+    elevation_image = ee.ImageCollection(dems).mosaic()
+    return ee.Image(elevation_image)
+
+def generate_elevation_map(dataset_list=None):
+    if not dataset_list:
+        dataset_list = ELEVATION_DATA.keys()
+
+    mosaic_image = mosaic_elevation_datasets(dataset_list)
+
+    final_image = visualize_elevation(mosaic_image)
+    data_params = DATASETS_VIS["projects/dgds-gee/bathymetry/gebco/2019"]
+    url = _get_gee_url(final_image)
+
     info = {}
-    info['dataset'] = 'gebco'
-    info['band'] = band
+    info['dataset'] = 'elevation'
+    info['band'] = 'elevation'
     linear_gradient = []
     palette = data_params['bathy_vis_params']['palette'] + data_params['topo_vis_params']['palette']
+    n_colors = len(palette)
+    offsets = np.linspace(0, 100, num=n_colors)
+    for color, offset in zip(palette, offsets):
+        linear_gradient.append({
+            'offset': '{:.3f}%'.format(offset),
+            'opacity': 100,
+            'color': color
+        })
+    info.update({
+        'url': url,
+        'linearGradient': linear_gradient,
+        'min': data_params['bathy_vis_params']['min'],
+        'max': data_params['topo_vis_params']['max'],
+        'imageId': None,
+        'datasets': list(dataset_list),
+        'function': 'mosaic_elevation_datasets'
+    })
+    return info
 
-    # TODO: call to _generate_image_info
+def visualize_gebco(source, band):
+    """
+    Specialized function to visualize GEBCO data
+    :param source: String, Google Earth Engine image id
+    :param band: String, band of image to visualize
+    :return: Dictionary
+    """
+    data_params = DATASETS_VIS[source]
+    image = ee.Image(source)
 
-    m = hillshaded.getMapId()
-    mapid = m.get('mapid')
-    token = m.get('token')
+    gebco = image.select(data_params['bandNames'][band])
+    land_mask = LANDMASK
 
-    url = 'https://earthengine.googleapis.com/map/{mapid}/{{z}}/{{x}}/{{y}}?token={token}'.format(
-        mapid=mapid,
-        token=token
-    )
+    hillshaded = visualize_elevation(gebco, land_mask, data_params)
+    url = _get_gee_url(hillshaded)
 
+    info = {}
+    info['dataset'] = 'gebco'
+    info['band'] = band
+    linear_gradient = []
+    palette = data_params['bathy_vis_params']['palette'] + data_params['topo_vis_params']['palette']
     n_colors = len(palette)
     offsets = np.linspace(0, 100, num=n_colors)
     for color, offset in zip(palette, offsets):
```

Comment:
I'm seeing this code part multiple times, can you make a create palette function?